public/app/plugins/datasource/loki/LogContextProvider.ts

```diff
@@ -0,0 +1,152 @@
+import { FieldCache, FieldType, LogRowModel, TimeRange, toUtc } from '@grafana/data';
+import { DataQuery } from '@grafana/schema';
+
+import LokiLanguageProvider from './LanguageProvider';
+import { LokiContextUi } from './components/LokiContextUi';
+import { REF_ID_STARTER_LOG_ROW_CONTEXT } from './datasource';
+import { escapeLabelValueInExactSelector } from './languageUtils';
+import { addLabelToQuery, addParserToQuery } from './modifyQuery';
+import { getParserFromQuery } from './queryUtils';
+import { ContextFilter, LokiQuery, LokiQueryDirection, LokiQueryType } from './types';
+
+export class LogContextProvider {
+  languageProvider: LokiLanguageProvider;
+  onContextClose: (() => void) | undefined;
+
+  constructor(languageProvider: LokiLanguageProvider) {
+    this.languageProvider = languageProvider;
+  }
+
+  prepareLogRowContextQueryTarget = async (
+    row: LogRowModel,
+    limit: number,
+    direction: 'BACKWARD' | 'FORWARD',
+    origQuery?: DataQuery
+  ): Promise<{ query: LokiQuery; range: TimeRange }> => {
```

Comment:
```suggestion
  async prepareLogRowContextQueryTarget(
    row: LogRowModel,
    limit: number,
    direction: 'BACKWARD' | 'FORWARD',
    origQuery?: DataQuery
  ): Promise<{ query: LokiQuery; range: TimeRange }> {
```

Just to be aligned with the other functions here.